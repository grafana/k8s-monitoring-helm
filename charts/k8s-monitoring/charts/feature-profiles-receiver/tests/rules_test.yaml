# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test profile relabeling rules
templates:
  - configmap.yaml
tests:
  - it: should render the correct configuration
    set:
      deployAsConfigMap: true
      profileProcessingRules: |
        // This creates a consistent hash value (0 or 1) for each unique combination of labels
        // Using multiple source labels provides better sampling distribution across your profiles
        rule {
          source_labels = ["env"]
          target_label = "__tmp_hash"
          action = "hashmod"
          modulus = 2
        }

        // This effectively samples ~50% of profile series
        // The same combination of source label values will always hash to the same number,
        // ensuring consistent sampling
        rule {
          source_labels = ["__tmp_hash"]
          action       = "drop"
          regex        = "^1$"
        }
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["module.alloy"]
