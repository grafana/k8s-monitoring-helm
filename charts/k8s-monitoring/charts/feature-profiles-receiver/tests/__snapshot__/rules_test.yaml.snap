should render the correct configuration:
  1: |
    |-
      declare "profiles_receiver" {
        argument "profiles_destinations" {
          comment = "Must be a list of profile destinations where collected profiles should be forwarded to"
        }

        pyroscope.receive_http "default" {
          http {
            listen_address = "0.0.0.0"
            listen_port = "4040"
          }

          forward_to = [pyroscope.relabel.default.receiver]
        }

        pyroscope.relabel "default" {
          // This creates a consistent hash value (0 or 1) for each unique combination of labels
          // Using multiple source labels provides better sampling distribution across your profiles
          rule {
            source_labels = ["env"]
            target_label = "__tmp_hash"
            action = "hashmod"
            modulus = 2
          }

          // This effectively samples ~50% of profile series
          // The same combination of source label values will always hash to the same number,
          // ensuring consistent sampling
          rule {
            source_labels = ["__tmp_hash"]
            action       = "drop"
            regex        = "^1$"
          }

          forward_to = argument.profiles_destinations.value
        }
      }
